src/s3access.cpp

```diff
@@ -110,27 +116,28 @@ QString S3Access::bucket() const
 
 S3ListResult *S3Access::list(QString const &prefix)
 {
-    char *date;
     QString signData;
     QString url;
+    s3_date date = now_as_s3date();
     QNetworkReply *reply;
     S3ListResult *result;
 
-    date = s3_make_date();
-    signData = "GET\n\n\n" + QString(date) + "\n/" + m_bucket + "/";
+    signData = "GET\n\n\n" + date + "\n/" + m_bucket + "/";
     url = "http://" + m_baseUrl + "/" + m_bucket + "/?delimiter=/";
     if (!prefix.isEmpty()) {
         url += "&prefix=" + prefix;
     }
 
     reply = performOp(GET, url, signData, date, nullptr, nullptr, nullptr);
     result = new S3ListResult(reply, this);
-    free(date);
 
     return result;
 }
 
-QNetworkReply *S3Access::performOp(Method method, QString const &url, QString const &sign_data, const char *date, QIODevice *in, const char *content_md5, const char *content_type)
+QNetworkReply *S3Access::performOp(Method method,
+                                   QString const &url, QString const &sign_data,
+                                   s3_date date, QIODevice *in,
+                                   const char *content_md5, const char *content_type)
```

Comment:
I know there's no offiical style guide or anything, and that it doesn't match the Qt styleguide either, but I'd prefer statements to all be kept on the same line.

Reply:
I look at the Qt styleguide later however the line was very long making it bad to read.

Reply:
https://wiki.qt.io/Qt_Coding_Style#Line_breaks

Reply:
Yes, what you've done is correctly in line with the Qt guidelines, but this is one area where the project diverges from them. If you have a look through the rest of the code you should see there's a "no-hard-linebreaks-mid-statement" rule applied.